errors/codes.go

```diff
@@ -8,11 +8,13 @@ const (
 	SpeechNilAudio = "SpeechNilAudio"
 	WAVCorruptFile = "WAVCorruptFile"
 	AudioFileNilStream = "AudioFileNilStream"
+	AudioFileNotWritableStream = "AudioFileNotWritableStream"
 )
 
 // errorMessages converts an error code to its corresponding message
 var errorMessages = map[ErrorCode]string{
 	SpeechNilAudio: "The audio file was nil. In order to convert a Speech object to Text, it must have a valid audio file. Usually, this means you created a Speech object that wasn't created using one of the Listen methods.",
 	WAVCorruptFile: "The wav file was corrupted. The wav file sent in does not have an incorrect header. This could be due to an incorrect file passed in or the header parameters were not fully passed in.",
-	AudioFileNilStream: "The audio stream was nil. Portaudio encountered an error in creating the file stream.",
+	AudioFileNilStream: "The audio stream was nil. Portaudio encountered an error in creating the file stream which may be due to an incorrectly formatted RIFF header. Check the file to make sure it was not corrupted or incomplete",
```

Comment:
1. `AudioFileNilStream` is misleading. The stream may or may not be nil, the error was that a stream could not be opened.
2. It isn't a "file stream". It's an output stream.
3. You're not passing the whole file to the function here. PortAudio returning an error is not due to a malformed file. The error will probably due to a speaker or microphone not connected, or permissions, or something related to accessing the actual hardware (might have to read up on this). Also, the header of the file is never passed in anyway, just the data, since we tell PortAudio separately the sample rate and sample size.